### DISCUSSION TITLE
Separate groups of params for different routes with overall persistance

### DISCUSSION THREAD

**Discussion by jschuur** [47ng/nuqs#602] [Ideas] [answered]
Not sure if this is possible easily now. I'd like to have several sets of parameters apply to different routes of the app, with the state of the others stored outside of the querystring when one group is active.

Example: `/path1?param1=abc&startDate=2024-01-01` and `/path2?param2=foo&startDate=2023-01-01`

Notice in this example `startDate` exists in both, but in my project I'd like to track separate start dates per group of params.

I know I can use `createSerializer` and swap out which group of params get applied to which path when I create the links. But is there an opportunity here for nuqs to offer up some kind of optional persistance layer beyond the URL?

I can just make a custom hook with a useEffect that looks for changes and writes them to context or another state managemet solution...

**Comment by franky47** [comment, marked as answer]
> is there an opportunity here for nuqs to offer up some kind of optional persistance layer beyond the URL?

Probably not, the scope of this package is to provide a declarative approach to URL state management, as a 1-1 mapping between the URL and state variables.

State updates can temporarily be decoupled from the URL (I just added a [demo of the deferred URL update](https://github.com/47ng/nuqs/blob/next/packages/e2e/src/app/app/deferred/page.tsx) pattern), but internal states will still be synced by key, so any `startDate` update would be propagated to any component watching that key.

It sounds like your "persistance layer" for non-selected sets of values would be best placed in an external state management solution (Zustand, Jotai, Context etc).
